### PR TITLE
Added variable autoscaling_memory to enable autoscaling based on AVG Memory

### PR DESCRIPTION
**Pull request by jrpradojr**

### CODE REVIEW

File: _variables.tf [hunk 1]
```diff
@@ -149,6 +154,11 @@ variable "autoscaling_target_cpu" {
   description = "Target average CPU percentage to track for autoscaling"
 }
 
+variable "autoscaling_target_mem" {
```
Comment (arthurbdiniz) [inline _variables.tf:157]: Maybe change this to `autoscaling_target_memory`.

File: appautoscaling.tf [hunk 2]
```diff
@@ -26,3 +26,23 @@ resource "aws_appautoscaling_policy" "scale_cpu" {
     }
   }
 }
+
+resource "aws_appautoscaling_policy" "scale_mem" {
```
Comment (arthurbdiniz) [inline appautoscaling.tf:30]: This resource name can be changed to `scale_memory`.

File: appautoscaling.tf [hunk 3]
```diff
@@ -26,3 +26,23 @@ resource "aws_appautoscaling_policy" "scale_cpu" {
...
+  count              = var.launch_type == "FARGATE" ? 0 : var.autoscaling_memory ? 1 : 0
+  name               = "scale-mem"
```
Comment (arthurbdiniz) [inline appautoscaling.tf:32]: This name can be changed to `scale-memory`.

File: appautoscaling.tf [hunk 4]
```diff
@@ -26,3 +26,23 @@ resource "aws_appautoscaling_policy" "scale_cpu" {
...
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_appautoscaling_target.ecs[0].resource_id
+  scalable_dimension = aws_appautoscaling_target.ecs[0].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.ecs[0].service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    target_value       = var.autoscaling_target_mem
```
Comment (arthurbdiniz) [inline appautoscaling.tf:39]: If you change the variable name, needs to be changed here as well.

**Review by arthurbdiniz — CHANGES_REQUESTED** [review]
Looks good! Just some changes for one variable and resource.

Also, I recommend running `terraform fmt --recursive` to make the lint pass for this PR.

File: appautoscaling.tf (hunk 4 above)
Comment (jrpradojr) [reply appautoscaling.tf]: Adjusted

File: appautoscaling.tf (hunk 3 above)
Comment (jrpradojr) [reply appautoscaling.tf]: Adjusted

File: appautoscaling.tf (hunk 2 above)
Comment (jrpradojr) [reply appautoscaling.tf]: Adjusted

**Review by arthurbdiniz — APPROVED** [review]

**Pull request merged**